### ISSUE TITLE
We need zip js for Angular 11

### ISSUE THREAD

**Issue by DJ-991** [gildas-lormeau/zip.js#474]
How can we use zip.js in Angular11? I need to encrypt zip having pdf. Please suggest

**Comment by gildas-lormeau** [comment]
What should be particular to Angular 11?

**Comment by DJ-991** [comment]
Just in angular how can we implement zip.js

**Comment by gildas-lormeau** [comment]
Like in any other JavaScript/TypeScript project?

**Comment by DJ-991** [comment]
This is angular project with typescript, we need to encrypt the zip having pdf blob. Can we do this with zip.js? 

Ask- whenever user open that downloaded zip, it should ask for the password.

**Comment by gildas-lormeau** [comment]
Yes, it is probably possible to implement this. What's blocking you?

**Comment by DJ-991** [comment]
After importing installing with npm and then importing in component.ts file. Which js file code from zip.js we can use ?

**Comment by DJ-991** [comment]
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/30df3599-d489-4a3b-872b-b38db1926de4)

**Comment by DJ-991** [comment]
PFA 
using like this : useCompressionStream is giving error in code
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/9b60197d-2ca2-4683-ae61-91b4e9e236d2)

**Comment by DJ-991** [comment]
that code we can use for encryption typescript code and for zipping ? @gildas-lormeau

**Comment by gildas-lormeau** [comment]
This is a code example that shows how to write a zip file with an AES password so I assume it's similar to what you want to do.

**Comment by DJ-991** [comment]
Will try this code hope this will work. @gildas-lormeau

**Comment by gildas-lormeau** [comment]
You'll probably have to follow these instructions as well: https://github.com/gildas-lormeau/zip.js/discussions/229

**Comment by DJ-991** [comment]
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/6a2e2133-7243-41b3-a387-1b715730260b)

1: Red color highlighted - this code we need to replace in angular.json file ? Yes/No 
2: Yellow color highlighted - need to create a file and paste this code there ? Yes/No 
3: Skyblue color highlighted - server like nothing is there, serve is there, is this code required ? Yes/No 

@gildas-lormeau pls help here

**Comment by DJ-991** [comment]
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/344d1b3a-1947-4fc1-ae64-2545de0add2a)
getting this error after config

here it is:
1: create file inside src - 
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/4c980379-85d5-4668-9d74-eea8e4d10a27)

2: angular.json "serve" config
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/2c48c363-29fb-4255-ab61-71787d174279)

3: angular.json "builder" and "options"
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/75566d52-7f73-47c5-9de9-67b7bc80a57b)

4: package.json
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/b4676d83-6e24-4bf0-b970-c4e8d03565ff)


My angular version is Angular 11. so i installed npm install -D @angular-builders/custom-webpack@11.1.1

**Comment by gildas-lormeau** [comment]
It looks like the ES version in your project is too old. Here is the fix of a similar issue in Angular CLI:
 - https://github.com/angular/angular-cli/commit/dd60228974a75b3c6b2ae98521444f0354161240
 - https://github.com/angular/angular-cli/issues/21196

**Comment by DJ-991** [comment]
in which file we need to change tsconfig.json or package.json ? not getting 
![image](https://github.com/gildas-lormeau/zip.js/assets/21131291/a24049db-9d82-4264-8e05-abcb23e457ee)

**Comment by gildas-lormeau** [comment]
I'm sorry but I don't know because I don't use Angular.
